liquidhaskell-boot/src/Language/Haskell/Liquid/Bare/Measure.hs

```diff
@@ -359,12 +364,92 @@ makeMeasureSpec env sigEnv myName (name, spec)
+makeOpaqueReflMeasures :: Bare.Env -> Bare.MeasEnv -> Bare.TycEnv -> Bare.ModSpecs ->
+              [(Ghc.Var, LocSpecType, F.Equation)] ->
+               Bare.Lookup ([MSpec SpecType Ghc.DataCon], [(Ghc.Var, Measure LocBareType ctor)])
+makeOpaqueReflMeasures env measEnv tycEnv specs eqs = do
+  res <- mapM transformVar $ S.toList varsToConsider
+  return $ unzip res
+  where
+    specsList = M.toList specs
+    -- Get all the defined symbols. Also, fully qualify them by the way (for those for which it's possible and not already done).
+    definedSymbols =
+      S.unions (uncurry getFromAxioms <$> specsList) -- reflections that ended up in equations
+      `S.union` getLocReflects (Just env) specs -- reflected symbols
+      `S.union` measVars -- Get the data constructors, ex. for Lit00.0
+      `S.union` S.unions (uncurry getDataDecls <$> specsList) -- get the Predicated type defs, ex. for T1669.CSemigroup
+      `S.union` S.unions (getAliases . snd <$> specsList) -- aliases, ex. for T1738Lib.incr
+    getFromAxioms modName spec = S.fromList $
+      Bare.qualifyLocSymbolTop env modName . localize . F.eqName <$> Ms.axeqs spec
+    measVars     = S.fromList $ localize . fst <$> getMeasVars env measEnv
+    getDataDecls modName spec = S.unions $
+      getFromDataCtor modName <$>
+        concat (tycDCons `Mb.mapMaybe` (dataDecls spec ++ newtyDecls spec))
+    getFromDataCtor modName decl = S.fromList $
+      Bare.qualifyLocSymbolTop env modName <$>
+        (dcName decl : (localize . fst <$> dcFields decl))
+    getAliases spec = S.fromList $ fmap rtName <$> Ms.ealiases spec
+    localize :: F.Symbol -> F.LocSymbol
+    localize sym = maybe (dummyLoc sym) varLocSym $ L.lookup sym (Bare.reSyms env)
+    -- Variables to consider
+    varsToConsider = S.unions $ uncurry3 (getOpaqueReflOfVar definedSymbols) <$> eqs
+    -- Main function: creates a (dummy) measure about a given variable
+    transformVar :: Ghc.Var -> Bare.Lookup (MSpec SpecType Ghc.DataCon, (Ghc.Var, Measure LocBareType ctor))
+    transformVar var = do
+      mspec <- mkMeasureDCon env (Bare.tcName tycEnv) (Ms.mkMSpec' [smeas])
+      return (mspec, (var, bmeas))
+      where
+        locSym = F.atLoc (loc specType) (F.symbol var)
+        specType = varSpecType var
+        bareType = varBareType var
+        bmeas = M locSym bareType [] MsReflect []
+        smeas = M locSym (val specType) [] MsReflect []
+
+-- Get the set of "free" symbols in the (reflection of the) unfolding of a given variable. Free symbols are those that are not
+-- already in the logic and that appear in the reflection of the unfolding but do not appear in the set of defined variables.
+-- For this purpose, you need to give a set of defined variables, the
+-- variable naming the definition to reflect and its corresponding
+-- equation in the logic.
+getOpaqueReflOfVar  :: S.HashSet LocSymbol -> Ghc.Var -> LocSpecType -> F.Equation -> S.HashSet Ghc.Var
```

Comment:
If definedSymbols where removed at the call site instead of here, `getOpaqueReflOfVar` could be renamed to `getFreeVarsOfReflectionOfVar`.